src/project_config.rs

```diff
@@ -0,0 +1,110 @@
+use serde::{Deserialize, Serialize};
+use snafu::Snafu;
+use std::path::{Path, PathBuf};
+
+#[derive(Debug, Snafu)]
+pub enum ProjectConfigError {
+    #[snafu(display("Unable to read config file {}: {}", path.display(), source))]
+    ReadFile {
+        source: std::io::Error,
+        path: PathBuf,
+    },
+    #[snafu(display("Unable to write config file {}: {}", path.display(), source))]
+    WriteFile {
+        source: std::io::Error,
+        path: PathBuf,
+    },
+    #[snafu(display("Unable to parse file {}: {}", path.display(), source))]
+    ParseFile {
+        source: toml::de::Error,
+        path: PathBuf,
+    },
+    #[snafu(display("The config file could not be serialized"))]
+    WriteToml {
+        source: toml::ser::Error,
+        path: PathBuf,
+    },
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+pub struct RenkuProjectConfig {
+    /// The version of this config file.
+    version: u16,
+
+    /// The base url to the renku platform.
+    pub renku_url: String,
```

Comment:
Same followup as the project URL. I think It is better to parse this and type it as a proper url. What do you think? Should we try to be more flexible?

Reply:
Yes totally agree!

Reply:
I now use the reqwest url parser to check the renku url